### PR TITLE
executor: migrate test-infra to testify for parts of executor_test

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref #28577

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/executor_issue_test.go [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+	require.Error(t, err)
+	require.Equal(t, "mock HashAggExec.baseExecutor.Open returned error", err.Error())
+	err = failpoint.Disable("github.com/pingcap/tidb/executor/mockHashAggExecBaseExecutorOpenReturnedError")
+	require.NoError(t, err)
+
+	// for StreamAggExec
+	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/executor/mockStreamAggExecBaseExecutorOpenReturnedError", `return(true)`))
+	_, err = tk.Exec("select sum(a) from (select 1 as a, 2 as b) t")
+	require.Error(t, err)
+	require.Equal(t, "mock StreamAggExec.baseExecutor.Open returned error", err.Error())
```
Comment (tisonkun) [inline executor/executor_issue_test.go:159]: You may replace these lines as:

```go
err = tk.ExecToErr(...)
require.EqualError(t, err, ...)
```

Comment (hawkingrei) [reply executor/executor_issue_test.go]: fixed all like these.

File: executor/executor_issue_test.go [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+	tk.MustExec("drop table if exists t;")
+	defer tk.MustExec("drop table if exists t;")
+	tk.MustExec("create table t(a int) partition by hash (a div 0) partitions 10;")
+	tk.MustExec("insert into t values (NULL);")
+
+	tk.MustQuery("select null div 0;").Check(testkit.Rows("<nil>"))
+	tk.MustQuery("select * from t;").Check(testkit.Rows("<nil>"))
+}
+
+func TestIssue24210(t *testing.T) {
```
Comment (tisonkun) [inline executor/executor_issue_test.go:133]: This test fails:

```
[2022-03-06T05:18:03.255Z] --- FAIL: TestIssue24210 (0.28s)
[2022-03-06T05:18:03.255Z]     executor_issue_test.go:156: 
[2022-03-06T05:18:03.255Z]         	Error Trace:	executor_issue_test.go:156
[2022-03-06T05:18:03.255Z]         	Error:      	Received unexpected error:
[2022-03-06T05:18:03.255Z]         	            	mock StreamAggExec.baseExecutor.Open returned error
[2022-03-06T05:18:03.255Z]         	            	github.com/pingcap/tidb/executor.(*StreamAggExec).Open
[2022-03-06T05:18:03.255Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/executor/aggregate.go:1237
[2022-03-06T05:18:03.255Z]         	            	github.com/pingcap/tidb/executor.(*ExecStmt).Exec
[2022-03-06T05:18:03.255Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/executor/adapter.go:407
[2022-03-06T05:18:03.255Z]         	            	github.com/pingcap/tidb/session.runStmt
[2022-03-06T05:18:03.255Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/session/session.go:1986
[2022-03-06T05:18:03.255Z]         	            	github.com/pingcap/tidb/session.(*session).ExecuteStmt
[2022-03-06T05:18:03.255Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/session/session.go:1863
[2022-03-06T05:18:03.255Z]         	            	github.com/pingcap/tidb/testkit.(*TestKit).Exec
[2022-03-06T05:18:03.255Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/testkit/testkit.go:215
[2022-03-06T05:18:03.255Z]         	            	github.com/pingcap/tidb/testkit.(*TestKit).ExecToErr
[2022-03-06T05:18:03.255Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/testkit/testkit.go:251
[2022-03-06T05:18:03.255Z]         	            	github.com/pingcap/tidb/executor_test.TestIssue24210
[2022-03-06T05:18:03.255Z]         	            		/home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/executor/executor_issue_test.go:154
[2022-03-06T05:18:03.255Z]         	            	testing.tRunner
[2022-03-06T05:18:03.255Z]         	            		/usr/local/go/src/testing/testing.go:1193
[2022-03-06T05:18:03.255Z]         	            	runtime.goexit
[2022-03-06T05:18:03.255Z]         	            		/usr/local/go/src/runtime/asm_amd64.s:1371
[2022-03-06T05:18:03.255Z]         	Test:       	TestIssue24210
[2022-03-06T05:18:03.255Z]         	Messages:   	<nil>
[2022-03-06T05:18:03.255Z] FAIL
```

Comment (hawkingrei) [reply executor/executor_issue_test.go:133]: Fixed

**Review by tisonkun — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**